sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -157,6 +157,38 @@ final class OffsetWindowFunctionFrame(
   override def currentUpperBound(): Int = throw new UnsupportedOperationException()
 }
 
+class FixedOffsetWindowFunctionFrame(
+    target: InternalRow,
+    ordinal: Int,
+    expressions: Array[OffsetWindowFunction],
+    inputSchema: Seq[Attribute],
+    newMutableProjection: (Seq[Expression], Seq[Attribute]) => MutableProjection,
+    offset: Int)
+  extends OffsetWindowFunctionFrame(
```

Comment:
See my comment below. I don't think you should extend this, if you need some functionality just put in in a common base class.

Reply:
I see.